### PR TITLE
return all documents when no source specified

### PR DESCRIPTION
**Pull request by jerpint**

### CODE REVIEW

File: buster/retriever/service.py [hunk 1]
```diff
@@ -32,13 +32,28 @@ def __init__(
...
+            # No source specified, return all documents
+            documents = self.db.documents.find()
+        else:
+            assert isinstance(source, str), "source must be a valid string."
+            source_id = self.get_source_id(source)
+
+            if source_id == "":
+                logger.warning(f"{source=} not found.")
+
+            documents = self.db.documents.find({"source_id": source_id})
```
Comment (hbertrand) [inline buster/retriever/service.py:54]: I don't know what happens if `source_id == ""` here. Maybe we should remove the filter and return all documents ?

File: buster/retriever/sqlite.py [hunk 2]
```diff
@@ -44,10 +44,10 @@ def __del__(self):
         if self.db_path is not None:
             self.conn.close()
 
-    def get_documents(self, source: str) -> pd.DataFrame:
+    def get_documents(self, source: str = None) -> pd.DataFrame:
         """Get all current documents from a given source."""
         # Execute the SQL statement and fetch the results.
-        if source == "":
+        if source is None:
```
Comment (hbertrand) [inline buster/retriever/sqlite.py:50]: You want to keep the `source == ""` here too.

**Review by hbertrand — CHANGES_REQUESTED** [review]

File: buster/retriever/sqlite.py (hunk 2 above)
Comment (jerpint) [reply buster/retriever/sqlite.py:50]: why ?

File: buster/retriever/service.py (hunk 1 above)
Comment (jerpint) [reply buster/retriever/service.py:54]: if source_id == "" it returns an empty list (i tried it)

Comment (jerpint) [reply buster/retriever/service.py:54]: i think its better to return an empty df than all of them

File: buster/retriever/sqlite.py (hunk 2 above)
Comment (jerpint) [reply buster/retriever/sqlite.py:50]: what happens if the source doesnt exist? does it also crash?

Comment (hbertrand) [reply buster/retriever/sqlite.py:50]: nope it just returns nothing

File: buster/retriever/service.py (hunk 1 above)
Comment (jerpint) [reply buster/retriever/service.py:54]: No they both do the same thing (both return empty dfs for an empty string passed)

**Review by hbertrand — APPROVED** [review]

**Pull request merged**